src/main/java/se/ams/dcatprocessor/parser/ApiDefinitionParser.java

```diff
@@ -38,57 +38,69 @@ public class ApiDefinitionParser {
+    };
+
     public static JSONObject getApiJsonString(String fileString) throws IOException, ParseException {
-        JSONParser parser = new JSONParser();
-        Stream<String> lines;
-        String apiJsonString = "";
+        JSONObject fullObj = parseToJsonObject(fileString);
+        JSONObject inner = getIn(fullObj, new String[]{"info", "x-dcat"});
+        return inner == null? fullObj : inner;
+    }
 
+    private static JSONObject getIn(JSONObject obj, String[] path) {
+        for (String k: path) {
+            obj = (JSONObject)obj.get(k);
+            if (obj == null) {
+                return null;
+            }
+        }
+        return obj;
+    }
+    
+    public static ApiSpecSyntax guessSyntax(String fileString) {
+        String trimmedString = fileString.trim();
+        if (trimmedString.startsWith("{") && trimmedString.endsWith("}")) {
+            return ApiSpecSyntax.Json;
+        }
+        
+        Stream<String> lines;
         lines = fileString.lines();
         String apiLine1 = lines.limit(1).collect(Collectors.joining("\n"));
 
-        if (apiLine1.contains("openapi")) {
-            apiJsonString = getFileApiYamlRaml(fileString);
-        } else if (apiLine1.contains("RAML")) {
-            apiJsonString = getFileApiYamlRaml(fileString);
-        } else if (apiLine1.contains("{")){
-            apiJsonString = fileString;
-        }
-        else {
-            throw new DcatException("not supported api definition");
+        if (apiLine1.contains("RAML") || apiLine1.contains("openapi")) {
+            return ApiSpecSyntax.YamlRaml;
         }
-        JSONObject jsonObjectFile = (JSONObject) parser.parse(apiJsonString);
-        if (jsonObjectFile.containsKey("info")) {
-            jsonObjectFile = (org.json.simple.JSONObject) (jsonObjectFile.get("info"));
-            jsonObjectFile = (org.json.simple.JSONObject) (jsonObjectFile.get("x-dcat"));
+
+        return ApiSpecSyntax.Unknown;
+    }
+    
+    private static JSONObject parseToJsonObject(String fileString) throws IOException, ParseException {
+        switch (guessSyntax(fileString)) {
+        case Json: return (JSONObject)((new JSONParser()).parse(fileString));
+        case YamlRaml: return convertYamlRamlToJsonObject(fileString);
+        case Unknown: {throw new DcatException("not supported api definition");}
         }
-        return jsonObjectFile;
+        return null;
     }
 
-    private static String getFileApiYamlRaml(String apiSpec) throws IOException {
-        FileOutputStream output = new FileOutputStream("output.json");
+    private static JSONObject convertYamlRamlToJsonObject(String apiSpec) throws IOException, ParseException {
+        FileOutputStream output = new FileOutputStream("output.json"); // TODO: Could we use a ByteArrayOutputStream for this, instead?
         JsonFactory factory = new JsonFactory();
         JsonGenerator generator = factory.createGenerator(output, JsonEncoding.UTF8);
 
-        try {
```

Comment:
Do I need to keep the `try-catch` here or can I throw it away? Cathing it and printing the stack trace potentially hides potential bugs in my opinion.

Reply:
Good and clear PR. I think it should be accepted immediately. We are trying to find a good working method for contributions. I just need to get the team to verify that the bug fixes and improvements are working for everyone. Is it Ok if it takes a few more days? Agree, don't hide exceptions. Remove the silent swallowing of the exception.

Reply:
Yes, no problem, these changes are not very urgent so take your time :-) I ran across the issues that these PRs address while using the DCAT-AP-SE-Processor from another code base and though I should fix them.